cluster-autoscaler/simulator/clustersnapshot/test_utils.go

```diff
@@ -53,3 +57,48 @@ func InitializeClusterSnapshotOrDie(
 		}
 	}
 }
+
+// CreateTestNodesWithPrefix creates n test Nodes with the given name prefix.
+func CreateTestNodesWithPrefix(prefix string, n int) []*apiv1.Node {
+	nodes := make([]*apiv1.Node, n, n)
+	for i := 0; i < n; i++ {
+		nodes[i] = test.BuildTestNode(fmt.Sprintf("%s-%d", prefix, i), math.MaxInt, math.MaxInt)
+		test.SetNodeReadyState(nodes[i], true, time.Time{})
+	}
+	return nodes
+}
+
+// CreateTestNodes creates n test Nodes.
+func CreateTestNodes(n int) []*apiv1.Node {
+	return CreateTestNodesWithPrefix("n", n)
+}
+
+// CreateTestPodsWithPrefix creates n test Pods with the given name prefix.
+func CreateTestPodsWithPrefix(prefix string, n int) []*apiv1.Pod {
+	pods := make([]*apiv1.Pod, n, n)
+	for i := 0; i < n; i++ {
+		pods[i] = test.BuildTestPod(fmt.Sprintf("%s-%d", prefix, i), 1, 1)
+	}
+	return pods
+}
+
+// CreateTestPods creates n test Pods.
+func CreateTestPods(n int) []*apiv1.Pod {
+	return CreateTestPodsWithPrefix("p", n)
+}
+
+// AssignTestPodsToNodes assigns test pods to test nodes based on their index position.
+func AssignTestPodsToNodes(pods []*apiv1.Pod, nodes []*apiv1.Node) {
+	if len(nodes) == 0 {
+		return
+	}
+
+	j := 0
+	for i := 0; i < len(pods); i++ {
+		if j >= len(nodes) {
```

Comment:
nit: `pods[i].Spec.NodeName = nodes[i%len(nodes)].Name` has the same effect, and does not need j variable to work.

Reply:
I agree this would be easier on my brain (although my suggestion about the func comment might be sufficient, for me at least)